src/Types/Link.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\GraphQL\Search\Types;
+
+use Atoolo\Resource\Resource;
+
+/**
+ * @codeCoverageIgnore
+ */
+class Link
+{
+    public function __construct(
+        public readonly string $url,
+        public readonly ?string $label = null,
+        public readonly ?string $ariaLabel = null,
+        public readonly ?string $description = null,
+        public readonly ?bool $opensNewWindow = null,
+        public readonly ?bool $isExternal = null,
+        public readonly ?Resource $resource = null,
```

Comment:
I don't know if it's a good idea to bind the link to a resource. The term link is very general and could also be useful elsewhere. Teasers can have these action links. You could perhaps still use this link object. Then we no longer need a `LinkResolver` but a `ResourceToLink` resolver (or something like that), which creates the link object completely.

Reply:
I now added a `ResourceLinkResolver` interface. (which is analogous to `ResourceDateResolver`, `ResourceKickerResolver` etc.)
It has to implementations `ArticleResourceLinkResolver` and `MediaResourceLinkResolver`, which are used by the teaser factories to resolve the link. 

I wonder if the teaser factory is the right place to resolve the link though. 
Should I move the link resolver from e.g. `ArticleTeaserFactory` to the `ArticleTeaserResolver`? Then again, what value does the `ArticleTeaserFactory` pass to the `link` field when calling the constrcutor of the `ArticleTeaser`?